peps/pep-9999.rst

```diff
@@ -0,0 +1,126 @@
+PEP: 9999
+Title: Static description file for Python installations
+Author: Filipe Laíns <lains@riseup.net>
+PEP-Delegate:
+Status: Draft
+Type: Informational
+Content-Type: text/x-rst
+Created: 01-Oct-2023
+Python-Version: 3.13
+Resolution:
+
+
+Abstract
+========
+
+Introduce a standard format for a static description file to describe Python
+installations.
+
+
+Rationale
+=========
+
+When introspecting a Python installation, running code is often undesirable or
+impossible. Having a static description file makes various of Python
+installation details available without having to run the interpreter.
+
+This is helpful for use-cases such as cross-compilation, Python launchers, etc.
+
+
+Scope
+=====
+
+This PEP only defines a standard format for a file describing Python
+installations, distributing such files is out of scope.
+
+Python implementations may choose to include a self-describing file as part of
+their distribution, but they are not required to, and it out of scope for this
+PEP to define how that may happen, if they decide to do so.
+
+
+Specification
+=============
+
+The standard Python installation description format consists of the JSON
+representation of a dictionary with the with the following keys.
+
+``schema_version``
+------------------
+
+:Type: ``number``
+:Description: Version of the schema to parse the file contents. It should be
+              ``1`` for the format described in this document. Future versions
+              may add, remove, or change fields.
+
+``language``
+------------
+
+Subsection with details related to the language specification.
+
+``version``
+~~~~~~~~~~~
+
+:Type: ``string``
+:Description: String representation the Python language version. Same as the
+              ``PY_VERSION`` macro on CPython.
+
+``version_parts``
+~~~~~~~~~~~~~~~~~
+
+:Type: ``object``
+:Description: Equivalent to :py:data:`sys.version_info`.
+
+``implementation``
+------------------
+
+Subsection with details related to Python implementation. Usually, this section
+should be roghly equivalent to :py:data:`sys.implementation`.
```

Comment:
```suggestion
should be roughly equivalent to :py:data:`sys.implementation`.
```

Reply:
Can you specify this in more detail? How is it roughy equivalent to `sys.implementation`, how is it not?

Reply:
It's implementation defined, but I want to make it recomended to be similar to `sys.implementation`.

Reply:
> Usually, this section
should be roghly equivalent to :py:data:`sys.implementation`.

How about changing this to:

> While only one key is required in this section, it SHOULD be equivalent to :py:data:`sys.implementation` on most implementations.

Reply:
I am gonna specify the "name" key is required, and the rest is implemention-defined.